### DISCUSSION TITLE
setPendingAdmin function

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#404] [Solidity / smart contracts] [answered]
In the NounsDAOExecutor there is the setPendingAdmin function. How is this function called since it requires the msg sender to be the contract itself? There is no other place in the smart contact that calls setPendingAdmin, does this mean in the current contract state that a new admin can't be set?

**Comment by solimander** [comment, marked as answer]
It can only be called by the Nouns DAO as the result of a successful proposal. In that case, the executor would call `setPendingAdmin` inside `executeTransaction`.
